### PR TITLE
Rust Improvements [PORT]

### PR DESCRIPTION
**Pull request by DoctorSquishy**
# About The Pull Request
Port of port from Bee:
"Rust is now an element (performance improvements)
Wirebrush now exists (clean rust)"
New Advanced Wirebrush sprite by me:
rustscrubbertest

## Why It's Good For The Game
Better performance.

image

## PR References

- https://github.com/BeeStation/BeeStation-Hornet/pull/6530
- https://github.com/BeeStation/BeeStation-Hornet/pull/6693

- https://github.com/tgstation/tgstation/pull/60496
- https://github.com/tgstation/tgstation/pull/60947
- https://github.com/tgstation/tgstation/pull/63789
- https://github.com/tgstation/tgstation/pull/59977

## Changelog

:cl: DoctorSquishy, Ivniinvi, Melbert, AnturK, ChungusGamer666, ZephyrTFA
code: Rust is now an element
add: Added Wirebrushes can be made at an autolathe to clean rust.
add: Added an advanced version  of the wirebrush, made at an autolathe, to quickly clean rust.
/:cl:

### CODE REVIEW

**Comment by DoctorSquishy** [comment]
> Does this conflict in any way with #360? I know it touches rusted walls.

I'll set this as draft and when bitmask is fully merged I'll. I'll check it out. I believe TG has both these systems implemented, so I don't think it will cause issues.

File: code/game/objects/items/tools/wirebrush.dm [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * If the person using this is unlucky they are going to die painfully.
+ */
+/obj/item/wirebrush/advanced
+	name = "advanced wirebrush"
+	desc = "An advanced wirebrush; uses radiation to almost instantly liquify rust."
+	icon_state = "wirebrush_adv"
+	toolspeed = 0.1
+
+	/// The amount of radiation to give to the user of this tool; regardless of what they did with it.
+	var/radiation_on_use = 20
```
Comment (nednaZ) [inline code/game/objects/items/tools/wirebrush.dm:24]: A lot of these variables could just be defines.

File: code/modules/research/designs/tool_designs.dm [hunk 2]
```diff
@@ -239,3 +239,24 @@
 	build_path = /obj/item/discovery_scanner
 	category = list("Tool Designs")
 	departmental_flags = DEPARTMENTAL_FLAG_SCIENCE
+
+/datum/design/wirebrush
+	name = "Wirebrush"
+	desc = "A tool to remove rust from walls."
```
Comment (nednaZ) [inline code/modules/research/designs/tool_designs.dm:245]:
```suggestion
	desc = "A tool to remove rust from surfaces."
```

**Review by nednaZ — CHANGES_REQUESTED** [review]
Just a couple minor changes needed, then I'll test merge.

File: code/game/objects/items/tools/wirebrush.dm [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+/obj/item/wirebrush/advanced/pre_attack(atom/A, mob/living/user)
+	. = ..()
+
+	if(!istype(user))
+		return
+
+	if(prob(crit_fail_prob))
+		to_chat(user, "<span class='danger'>You feel a sharp pain as your entire body grows oddly warm.</span>")
+		user.radiation += crit_fail_rads
```
[the remark this answers is not in the document]
Comment (DoctorSquishy) [reply code/game/objects/items/tools/wirebrush.dm:50]: I found a solution, I guess the original coder did not realize about `rad_act ` So now rad protections works with this brush.

**Comment by DoctorSquishy** [comment]
I'm gonna redo the icon for at least the advanced brush. I feel like it needs to feel a little more advanced

**Comment by DoctorSquishy** [comment]
Wirebrush code further improved and new icon for the advanced wirebrush

File: code/game/objects/items/tools/wirebrush.dm [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		return
+
+	user.rad_act(RADIATION_ON_USE) //Apply rads on user
+
+	if(user.radiation > 100) //To warn the user
+		if(prob(10))
+			to_chat(user, "<span class='danger'>You feel an odd warm tingling sensation coming from the brush.</span>")
+
+	if(prob(CRIT_FAIL_PROB))
+		if(HAS_TRAIT(user, TRAIT_RADIMMUNE)) // For those with radimmunity
```
Comment (nednaZ) [inline code/game/objects/items/tools/wirebrush.dm:52]: Shouldn't this be checking if they DON'T have radimmune?

Comment (DoctorSquishy) [reply code/game/objects/items/tools/wirebrush.dm:52]: It is so those with those with TRAIT_RADIMMUNE see a little feedback and get rads even on a crit fail so they too can potentially glow green from using this, I basically wanted them to not get the other message with sharp pain and vomit

**Comment by nednaZ** [comment]
Testmerging to confirm we have no issues

**Comment by DoctorSquishy** [comment]
Probably could expand on the rust overlays later to add some variety with rust

**Review by nednaZ — APPROVED** [review]
I have seen zero issues during test merging.

**Pull request merged**